src/UnderlineNav2/styles.ts

```diff
@@ -92,15 +92,23 @@ export const getLeftArrowHiddenBtn = (theme?: Theme) => ({
   opacity: 0,
   top: 0,
   bottom: 0,
-  left: 0
+  left: 0,
+  /* This is intentionally set to prevent any visual focus ring from showing up
+   * when a hidden element gets focus just to trigger assistive technology to reread */
+  '&[aria-disabled="true"]:focus': {
+    outline: 'none'
+  }
```

Comment:
```suggestion
  left: 0
```

This might seem counter-intuitive at first, but we'll actually want to preserve the focus ring. The control is still interactive in this context in that it communicates being conditionally disabled. 

A user who relies on focus rings to understand what is and is not interactive would need the ring preserved here, with the button's visuals (and potential announcement) communicating conditionally-applied state.

Reply:
Hi @ericwbailey 👋🏼 Thank you so much for your review! I recorded a quick video to reply here because needed to demonstrate something, I hope it is helpful! Let me know what you think 😊 

https://user-images.githubusercontent.com/1446503/194270894-c1e00a46-0f5d-4c46-b5e9-d4947dde9312.mp4

Reply:
Oh, this is interesting. Is there a place I can test this behavior?

Reply:
Yes, this PR's [storybook link](https://primer-9cc12887c1-13348165.drafts.github.io/storybook/?path=/story/components-underlinenav--internal-responsive-nav) should work. Thank you!!

Reply:
Just for providing more context here, it happens because we eventually hide the arrow button when there is no overflow on the direction that the button controls. And we add `aria-disabled=true` and re-focus on it just before hiding it. I hope it helps!

Reply:
👋🏼 again @ericwbailey! I pushed some update on the buttons since my last message above and this behaviour doesn't really occur anymore which is good I believe? I added a screen recording on the description and the [storybook link](https://primer-9cc12887c1-13348165.drafts.github.io/storybook/?path=/story/components-underlinenav--internal-responsive-nav) has the updated version. It doesn't occur anymore because we used to hide (opacity:0) the buttons with a `transition` but now we do `display:none` with a `transform:translateX(dynamicValue)`  when there is no overflow but the rest is same as in adding `aria-disable=true` and refocusing. I am sorry for pushing new changes; we didn't really have many alternatives to solve [this specific problem](https://github.com/primer/react/pull/2406#issuecomment-1271307239) 😞 Looking forward to hearing what you think!

Reply:
Thanks for the detail for the changes! I just re-tested it on VoiceOver on iOS and it sounds like it works as intended 🎉